presto-main/src/test/java/com/facebook/presto/server/TestServer.java

```diff
@@ -130,12 +135,64 @@ public void testQuery()
                 data.addAll(queryResults.getData());
             }
         }
+        assertNull(queryResults.getError());
 
         // only the system catalog exists by default
         List<List<Object>> rows = data.build();
         assertEquals(rows, ImmutableList.of(ImmutableList.of("system")));
     }
 
+    @Test
+    public void testTransactionSupport()
+            throws Exception
+    {
+        Request request = preparePost()
+                .setUri(uriFor("/v1/statement"))
+                .setBodyGenerator(createStaticBodyGenerator("start transaction", UTF_8))
+                .setHeader(PRESTO_USER, "user")
+                .setHeader(PRESTO_SOURCE, "source")
+                .setHeader(PRESTO_TRANSACTION_ID, "none")
+                .build();
+
+        QueryResults queryResults = client.execute(request, createJsonResponseHandler(jsonCodec(QueryResults.class)));
+        ImmutableList.Builder<List<Object>> data = ImmutableList.builder();
+        while (true) {
+            if (queryResults.getData() != null) {
+                data.addAll(queryResults.getData());
+            }
+
+            if (queryResults.getNextUri() == null) {
+                break;
+            }
+            queryResults = client.execute(prepareGet().setUri(queryResults.getNextUri()).build(), createJsonResponseHandler(jsonCodec(QueryResults.class)));
+        }
+        assertNull(queryResults.getError());
+
+        // only the system catalog exists by default
+        List<List<Object>> rows = data.build();
+        assertEquals(rows, ImmutableList.of(ImmutableList.of(TRUE)));
```

Comment:
We don't actually care about the data for this. It's a dummy row to make some clients happy. Instead, we need to verify that we got the `PRESTO_STARTED_TRANSACTION_ID` header.